GPU/Common/SplineCommon.cpp

```diff
@@ -361,16 +383,45 @@ static void _SplinePatchFullQuality(u8 *&dest, u16 *indices, int &count, const S
 	}
 }
 
+template <bool origNrm, bool origCol>
+static inline void SplinePatchFullQualityDispatch3(u8 *&dest, u16 *indices, int &count, const SplinePatchLocal &spatch, u32 origVertType, int quality, int maxVertices) {
+	bool origTc = (origVertType & GE_VTYPE_TC_MASK) != 0;
```

Comment:
Ahh, I was just misunderstanding that is loop unrolling...
It was just separate and duplicating at if-branch in loops using templates.
Anyway, I didn't know this template dispatch chain, and it seems strange, but effective in hot loops.
I'm not thinking so bigger rewrite, but I hope to more readable and performance improved.

BTW, did you use any tools for profiling? I don't know how to profile and optimize well. VS 2015 has performance profiler, I should learn how to use this maybe.